src/declarativebookmarkmodel.cpp

```diff
@@ -47,12 +51,12 @@ void DeclarativeBookmarkModel::removeBookmark(const QString& url)
         return;
     }
 
-    if (bookmarks.contains(url)) {
-        int index = bookmarkUrls.indexOf(url);
+    if (bookmarkIndexes.keys().contains(url)) {
```

Comment:
Basically the performace is the only reason why I decided to stick to QMap: O(N) vs O(log2 N).
Do we want to sacrifice that?